chrome/content/zotero/xpcom/sync/syncRunner.js

```diff
@@ -231,6 +237,39 @@ Zotero.Sync.Runner_Module = function (options = {}) {
 		
 		return json;
 	});
+
+
+	// Prompt if library empty and there is no userID stored
+	this.checkEmptyLibrary = Zotero.Promise.coroutine(function* (keyInfo) {
+		let library = Zotero.Libraries.userLibrary;
+		let userID = Zotero.Users.getCurrentUserID();
+		let hasItems = yield library.hasItems();
```

Comment:
This should be a very quick query, but still no reason to run it unless `!userID`.